nixos/modules/system/boot/systemd.nix

```diff
@@ -746,6 +746,86 @@ in
+            };
+            enable = mkOption {
+              type = types.bool;
+              default = true;
+              description = "disable this tmpfiles rule";
+            };
+            type = mkOption {
+              type = types.str;
+              description = ''
+                The type consists of a single letter and optionally an exclamation mark ("!") and/or minus sign ("-").
+                See <citerefentry><refentrytitle>tmpfiles.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+                for the exact format.
+              '';
+            };
+            mode = emptyOption ''
+              The file access mode to use when creating this file or directory.
+              If omitted or when set to "-", the default is used: 0755 for directories,
+              0644 for all other file objects. For z, Z lines, if omitted or when set
+              to "-", the file access mode will not be modified. This parameter is ignored for x, r, R, L, t, and a lines.
+            '';
+            user = emptyOption ''
+              The user and group to use for this file or directory.
+              This may either be a numeric ID or a user/group name.
+              If omitted or when set to "-", the user and group of the user who invokes systemd-tmpfiles is used.
+            '';
+            group = emptyOption ''
+              The user and group to use for this file or directory.
+              This may either be a numeric ID or a user/group name.
+              If omitted or when set to "-", the user and group of the user who invokes systemd-tmpfiles is used.
+            '';
+            age = emptyOption ''
+              The date field, when set, is used to decide what files to delete when cleaning.
+              If a file or directory is older than the current time minus the age field, it is deleted.
+              The field format is a series of integers each followed by one of the following suffixes
+              for the respective time units: s, m or min, h, d, w, ms, and us, meaning seconds, minutes,
+              hours, days, weeks, milliseconds, and microseconds, respectively. Full names of the time units can be used too.
+            '';
+            arguments = emptyOption ''
+              For L lines determines the destination path of the symlink.
+              For c and b, determines the major/minor of the device node, with major and minor formatted as integers,
+              separated by ":", e.g. "1:3". For f, F, and w, the argument may be used to specify a short string that
+              is written to the file, suffixed by a newline. For C, specifies the source file or directory.
+              For t and T, determines extended attributes to be set. For a and A, determines ACL attributes to be set.
+              For h and H, determines the file attributes to set. Ignored for all other lines.
```

Comment:
Maybe all of these descriptions should be a link to upstream's docs? For instance, we can use links such as:

- https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html#Type
- https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html#Path
- https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html#Mode

etc.